pkg/kubelet/kubelet.go

```diff
@@ -1938,6 +1949,7 @@ func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 func (kl *Kubelet) HandlePodUpdates(pods []*v1.Pod) {
 	start := kl.clock.Now()
 	for _, pod := range pods {
+		kl.secretManager.RegisterPod(pod)
```

Comment:
do we need to call this in HandlePodReconcile as well? I'm concerned that we'll miss registering a pod parallel to all the places we add/update/remove it in the podManager